src/distrs/beta.jl

```diff
@@ -1,19 +1,71 @@
 # functions related to beta distributions
 
+using HypergeometricFunctions: _₂F₁
+
 import .RFunctions:
-    betapdf,
-    betalogpdf,
-    betacdf,
-    betaccdf,
-    betalogcdf,
-    betalogccdf,
-    betainvcdf,
-    betainvccdf,
+    # betapdf,
+    # betalogpdf,
+    # betacdf,
+    # betaccdf,
+    # betalogcdf,
+    # betalogccdf,
+    # betainvcdf,
+    # betainvccdf,
     betainvlogcdf,
     betainvlogccdf
 
 # pdf for numbers with generic types
-betapdf(α::Real, β::Real, x::Number) = x^(α - 1) * (1 - x)^(β - 1) / beta(α, β)
+betapdf(α::Real, β::Real, x::Real) = exp(betalogpdf(α, β, x))
 
 # logpdf for numbers with generic types
-betalogpdf(α::Real, β::Real, x::Number) = (α - 1) * log(x) + (β - 1) * log1p(-x) - logbeta(α, β)
+function betalogpdf(α::T, β::T, x::T) where T<:Real
+    return if x <= 0.5
+        xlogy(α - 1, x) + (β - 1) * log1p(-x) - logbeta(α, β)
+    else
+        betalogpdf(β, α, 1 - x)
+    end
+end
+betalogpdf(α::Real, β::Real, x::Real) = betalogpdf(promote(α, β, x)...)
+
+betacdf(α::Float64, β::Float64, x::Float64) = first(beta_inc(α, β, x))
+betacdf(α::Real, β::Real, x::Real) = betacdf(promote(float(α), β, x)...)
+betacdf(k::T, x::T) where T = throw(MethodError(betacdf, (α, β, x)))
```

Comment:
Is it actually necessary to add these `MethodErrors`? A `MethodError` would be returned anyways it seems? Also, is this supposed to be
```suggestion
betacdf(α::T, β::T, x::T) where T = throw(MethodError(betacdf, (α, β, x)))
```
?

Reply:
This is to avoid a stack overflow for e.g. `Float32` elements.

Reply:
I see. Would it maybe be easier to just rely on the promotions in SpecialFunctions in https://github.com/JuliaMath/SpecialFunctions.jl/blob/fbb44faaa06f7b05723ac38811f7786d3142e0b5/src/beta_inc.jl#L880-L884 and call the four-argument version (unfortunately, it seems https://github.com/JuliaMath/SpecialFunctions.jl/blob/fbb44faaa06f7b05723ac38811f7786d3142e0b5/src/beta_inc.jl#L879 is only defined for `Float64` - I guess it should be generalized to `Real`?)?